otherlibs/stdune/src/copyfile_stubs.c

```diff
@@ -0,0 +1,40 @@
+#include <caml/memory.h>
+#include <caml/mlvalues.h>
+
+#if defined(__APPLE__)
+#include <caml/threads.h>
+#include <caml/unixsupport.h>
+#include <copyfile.h>
+
+CAMLprim value stdune_copyfile(value v_from, value v_to) {
+  CAMLparam2(v_from, v_to);
+  const char *from = (void *)caml_stat_strdup(String_val(v_from));
+  const char *to = (void *)caml_stat_strdup(String_val(v_to));
+  caml_release_runtime_system();
+  int ret = copyfile(from, to, NULL, COPYFILE_CLONE);
+  caml_stat_free((void *)from);
+  caml_stat_free((void *)to);
```

Comment:
Every pointer type is implicitly compatible with `void *`.
```suggestion
  caml_stat_free(from);
  caml_stat_free(to);
```

Reply:
Without them I get these pesky warnings:

```
copyfile_stubs.c:15:18: warning: passing 'const char *' to parameter of type 'caml_stat_block' (aka 'void *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  caml_stat_free(from);
                 ^~~~
/nix/store/3p4yg2gbnmv6l2yj0gff4hm6wxcsa95p-ocaml-base-compiler-4.14.0/lib/ocaml/caml/memory.h:136:47: note: passing argument to parameter here
CAMLextern void caml_stat_free(caml_stat_block);
                                              ^
copyfile_stubs.c:16:18: warning: passing 'const char *' to parameter of type 'caml_stat_block' (aka 'void *') discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  caml_stat_free(to);
                 ^~
/nix/store/3p4yg2gbnmv6l2yj0gff4hm6wxcsa95p-ocaml-base-compiler-4.14.0/lib/ocaml/caml/memory.h:136:47: note: passing argument to parameter here
CAMLextern void caml_stat_free(caml_stat_block);
                                              ^
```

Reply:
The compiler is right: your `to` and `from` values cannot be `const` pointers because you pass them to `caml_stat_free`. The "right" solution would be to remove the `const` in my opinion, instead of introducing explicit casts.

Reply:
done